Lib/test/test_capi.py

```diff
@@ -963,6 +977,8 @@ def some(pos_only='p', zero=0, optional=None):
 
         with self.assertRaises(SystemError):
             _testcapi.function_set_defaults(some, 1)  # not tuple or None
```

Comment:
It would be good to check here that a failed set in fact did not change the defaults.

Reply:
Ah I see this is checked after the next failed set. But better to check every time.